service/handlers.go

```
@@ -838,6 +859,92 @@ func (h handler) listTargets(w http.ResponseWriter, r *http.Request) {
+	ah := r.Header.Get("Authorization")
+	a, err := credentials.NewAuthorization(ah)
+	if err != nil {
+		h.errorResponse(w, "error unauthorized, invalid authorization header format", http.StatusUnauthorized)
+		return
+	}
+	if err := a.Validate(a.ValidateAuthorizedAdmin(h.env.AdminSecret)); err != nil {
+		h.errorResponse(w, "unauthorized", http.StatusUnauthorized)
+		return
+	}
+	level.Debug(l).Log("message", "reading request body")
+
+	var utr requests.UpdateTarget
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		level.Error(l).Log("message", "error reading request data", "error", err)
+		h.errorResponse(w, "error reading request data", http.StatusInternalServerError)
+	}
+
+	if err := json.Unmarshal(reqBody, &utr); err != nil {
+		level.Error(l).Log("message", "error processing request", "error", err)
+		h.errorResponse(w, "error processing request", http.StatusBadRequest)
+		return
+	}
+
+	if err := utr.Validate(); err != nil {
+		level.Error(l).Log("message", "error invalid request", "error", err)
+		h.errorResponse(w, fmt.Sprintf("invalid request, %s", err), http.StatusBadRequest)
+		return
+	}
+
+	l = log.With(l, "target", targetName)
+
+	level.Debug(l).Log("message", "creating credential provider")
+	cp, err := h.newCredentialsProvider(*a, h.env, r.Header, credentials.NewVaultConfig, credentials.NewVaultSvc)
+	if err != nil {
+		level.Error(l).Log("message", "error creating credentials provider", "error", err)
+		h.errorResponse(w, "error creating credentials provider", http.StatusInternalServerError)
+		return
+	}
+
+	projectExists, err := cp.ProjectExists(projectName)
+	if err != nil {
+		level.Error(l).Log("message", "error determining if project exists", "error", err)
+	}
+
+	// TODO Perhaps this should be 404
+	if !projectExists {
+		level.Error(l).Log("message", "project does not exist")
+		h.errorResponse(w, "project does not exist", http.StatusBadRequest)
+		return
+	}
+
+	targetExists, _ := cp.TargetExists(projectName, targetName)
```

I think we should handle the error instead of swallowing it